### ISSUE TITLE
Add typed clients/informers/listers for Sagemaker CRDs

### ISSUE THREAD

**Issue by song-william** [aws/amazon-sagemaker-operator-for-k8s#120]
**What would you like to be added**:
In addition to the generated `DeepCopy` functions, may this project also include generated code for [clients](https://github.com/kubernetes/code-generator/tree/master/cmd/client-gen), [informers](https://github.com/kubernetes/code-generator/tree/master/cmd/informer-gen), and [listers](https://github.com/kubernetes/code-generator/tree/master/cmd/lister-gen)? The documentation on using the generators is a bit sparse, but I have found this [blog post by Stefan Schimanski](https://www.openshift.com/blog/kubernetes-deep-dive-code-generation-customresources) to be a good resource on using the scripts. There is also a master script ([generator-group.sh](https://github.com/kubernetes/code-generator/blob/master/generate-groups.sh)) that can be used to generate all the code at once.

**Why is this needed**:

This will enable a native/full-featured experience for users to programmatically interact with Sagemaker's CRDs. Otherwise, users have to rely on dynamic clients/informers and type casting, which is a bit more error prone and inconvenient.

**Comment by RedbackThomson** [comment]
I have spent the last few days working on generating these typings using the links provided, but ultimately were not able to get them working. It seems that these scripts are either out of date or simply have a bug that prevents them from producing compatible outputs.

The `kubernetes/code-generator` code will not work with multiple groups. I have created a ticket to address this - https://github.com/kubernetes/code-generator/issues/110 . Even if it is compatible it would require changing our directory structure from `api/v1/<group>` to `api/<group>/v1` and all the imports would change with it (requiring a major version update). 

We would recommend using dynamic clients and informers. Kubebuilder recommends using these over static types - https://github.com/kubernetes-sigs/kubebuilder/issues/1152#issuecomment-584774562 - and have said they are unlikely to add native support for informers or client sets.
